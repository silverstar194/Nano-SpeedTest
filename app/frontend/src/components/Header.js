import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
    <div className="nav nav-tabs">
        <li className="nav-item">
            <Link className="nav-link" to=''>Speed Test</Link>
        </li>
        <li className="nav-item">
            <Link className="nav-link" to='Stats'>Statistics</Link>
        </li>
        <li className="nav-item">
            <Link className='nav-link' to='Info'>More Info</Link>
        </li>
    </div>
);

export default Header;