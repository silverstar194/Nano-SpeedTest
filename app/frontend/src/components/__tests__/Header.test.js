import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    navigation: {
        activeTab: ''
    }
});

it('shallow renders without crashing', () => {
    shallow(<Header store={store}/>);
});

it(`renders snapshot correctly when activeTab is ''`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <Header />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});