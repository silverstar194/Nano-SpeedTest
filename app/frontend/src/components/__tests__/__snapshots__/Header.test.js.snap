// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly when activeTab is '' 1`] = `
<div
  className="nav nav-tabs"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link active"
      href="/"
      onClick={[Function]}
    >
       
      Speed Test
       
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link "
      href="/Stats"
      onClick={[Function]}
    >
       
      Statistics
       
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link "
      href="/Info"
      onClick={[Function]}
    >
       
      More Info
       
    </a>
  </li>
</div>
`;
